src/utils/warnings.ts

```diff
@@ -6,22 +6,30 @@
 /** A {@link WarningHandler} is called when a new warning is emitted */
 export type WarningHandler = (message: string) => void;
```

Comment:
I think this should take the timeout as well, making the Warnings class the only one dealing with timeout (and then handlers can decide to ignore it)

```suggestion
export type WarningHandler = (message: string, timeout: number) => void;
```

Reply:
done